### PR TITLE
Subworkflow doesn't return output

### PR DESCRIPTION
**Pull request by sukruthiarjula**
Subworkflow doesn't return output #1849

### CODE REVIEW

File: core/src/main/java/com/netflix/conductor/core/execution/tasks/SubWorkflow.java [hunk 1]
```diff
@@ -114,7 +114,10 @@ public void start(Workflow workflow, Task task, WorkflowExecutor provider) {
 			task.getOutputData().put(SUB_WORKFLOW_ID, subWorkflowId);
 
 			// Set task status based on current sub-workflow status, as the status can change in recursion by the time we update here.
-			Workflow subWorkflow = provider.getWorkflow(subWorkflowId, false);
+			Workflow subWorkflow = provider.getWorkflow(subWorkflowId, false);			
+			if (subWorkflow.getStatus() == WorkflowStatus.COMPLETED) {
```
Comment (pweatherby-ccctc) [inline core/src/main/java/com/netflix/conductor/core/execution/tasks/SubWorkflow.java:118]: @sukruthiarjula While this definitely gets the job done for the best-case scenario, there are other cases it misses. Consider how the code does this check/copy over of the payload as seen on lines 135 to 142.

Using the isTerminal function will cover more scenarios than just the COMPLETED scenario and then making sure to account for external payload storage is crucial as well.

Perhaps those lines (135-142) can be refactored out into a function of its own to be called by both the execute function and the start function. just a thought.

**Review by pweatherby-ccctc — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/netflix/conductor/core/execution/tasks/SubWorkflow.java (hunk 1 above)
Comment (sukruthiarjula) [reply core/src/main/java/com/netflix/conductor/core/execution/tasks/SubWorkflow.java]: Hi @pweatherby-ccctc,

As discussed, I have made the mentioned changes.

**Review by pweatherby-ccctc — APPROVED** [review]
Thank you so much for putting this together. Unfortunately, the build is now failing on a Logging Metric test. so once that gets looked at and adjusted, I think this will be good to go.

**Comment by sukruthiarjula** [comment]
Build is failing on a Logging Metric test due to timeout issue. In local I could not replicate as all the 77 tests passed.
image

The code coverage is failing in CassandraMetadataDAO.java unrelated to the changes which I have made. Any inputs provided would be helpful.

**Review by apanicker-nflx — APPROVED** [review]

**Pull request merged**